addons/mrp/wizard/change_production_qty.py

```diff
@@ -92,6 +92,9 @@ def change_prod_qty(self):
                                  operation.workcenter_id.time_stop +
                                  cycle_number * operation.time_cycle * 100.0 / operation.workcenter_id.time_efficiency)
                 quantity = wo.qty_production - wo.qty_produced
+                is_not_first_order = bool(production.workorder_ids.filtered(lambda l: l.next_work_order_id == wo))
+                if wo.operation_id.batch and wo.state in ['processing', 'ready'] and is_not_first_order:
+                    quantity = wo.qty_producing
```

Comment:
You could probably call _defaults_from_finished_workorder_line. Anyway this would be erase 3 lines after since you didn't use elif. Check if we can get rid of this.